docs/apworld_dev_faq.md

```diff
@@ -43,3 +43,25 @@ A faster alternative to the `for` loop would be to use a [list comprehension](ht
 ```py
 item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
 ```
+
+---
+
+### I learned about indirect conditions in the world API document, but I want to know more. What are they and why are they necessary?
+
+The world API document mentions indirect conditions and **when** you should use them, but not *how* they work and *why* they are necessary. This is because the explanation is quite complicated.
+
+Region sweep (the algorithm that determines which regions are reachable) is a Breadth First Search of the region graph from the Menu region, checking entrances one by one and adding newly reached nodes (regions) and their entrances to the queue until there is nothing more to check.
+
+For performance reasons AP only checks every entrance once. However, if entrance access conditions depend on regions, then it is possible for this to happen:
+1. An entrance that depends on a region is checked and determined to be nontraversable because the region hasn't been reached yet during the graph search.
+2. After that, the region is reached by the graph search. The entrance *would* now be determined to be traversable if it were rechecked.
+
+To account for this case, AP would have to recheck all entrances every time a new region is reached, until no new regions are reached.
```

Comment:
```suggestion
To account for this case, AP would have to recheck all entrances every time a new region is reached until no new regions are reached.
```

Reply:
Is this comma actually correct?

Reply:
Removing the comma is fine, it's also technically fine to keep. The sentence is kind of awkward but I'm not really thinking of anything better